### PR TITLE
EIP E2Es: add non-OVN managed network test cases for enable/disable e…

### PR DESCRIPTION
**Pull request by martinkennelly**
…gress for nodes / pods

See commit for more details.

### CODE REVIEW

**Comment by martinkennelly** [comment]
Waiting on static route improvements from @tssurya 
Tested with https://github.com/ovn-org/ovn-kubernetes/pull/3885 and its ok.

File: test/e2e/egressip.go [hunk 1]
```diff
@@ -1591,4 +1622,502 @@ spec:
...
+			}
+		}
+		gomega.Expect(nodeNameHostingOVNManagedEIP).ShouldNot(gomega.BeEmpty())
+		gomega.Expect(nodeNameHostingNonOVNManagedEIP).ShouldNot(gomega.BeEmpty())
+		egressNodeAvailabilityHandler.Disable(nodeNameHostingOVNManagedEIP)
+
+		ginkgo.By("15. Check that the status is of length one")
+		statuses = verifyEgressIPStatusLengthEquals(1, nil)
+
+		ginkgo.By("16. Check connectivity both pods to an external \"node\" on the non-OVN managed network and " +
```
Comment (jcaamano) [inline test/e2e/egressip.go:2057]: nit: from

File: test/e2e/egressip.go [hunk 2]
```diff
@@ -1591,4 +1622,502 @@ spec:
...
+			} else if status.EgressIP == egressIPNonOVNManaged {
+				nodeNameHostingNonOVNManagedEIP = status.Node
+			}
+		}
+		gomega.Expect(nodeNameHostingOVNManagedEIP).ShouldNot(gomega.BeEmpty())
+		gomega.Expect(nodeNameHostingNonOVNManagedEIP).ShouldNot(gomega.BeEmpty())
+		egressNodeAvailabilityHandler.Disable(nodeNameHostingOVNManagedEIP)
+
+		ginkgo.By("15. Check that the status is of length one")
+		statuses = verifyEgressIPStatusLengthEquals(1, nil)
```
Comment (jcaamano) [inline test/e2e/egressip.go:2055]: should we check that the status has the egress ip that we expect here and in similar other steps?

File: test/e2e/egressip.go [hunk 3]
```diff
@@ -25,6 +24,9 @@ const (
 	DEFAULT_OVN_EGRESSIP_GRPC_HEALTHCHECK_PORT = "9107"
 	OVN_EGRESSIP_LEGACY_HEALTHCHECK_PORT_ENV   = "0" // the env value to enable legacy health check
 	OVN_EGRESSIP_LEGACY_HEALTHCHECK_PORT       = "9" // the actual port used by legacy health check
+	secondaryIPV4Subnet                        = "10.10.10.0/24"
+	secondaryIPV6Subnet                        = "2001:db8:abcd:1234:c000::/64"
+	secondaryNetworkName                       = "secondary-ovn"
```
[the remark this answers is not in the document]
Comment (tssurya) [reply test/e2e/egressip.go:29]: I have a more fundamental question, why is this feature called "non-ovn-managed" when clearly it uses OVN to do the rerouting?
are we just meaning "the-network-that-is-not-shared-with-br-ex"?

[the remark this answers is not in the document]
Comment (martinkennelly) [reply test/e2e/egressip.go:29]: It does part of the routing yes (out the mgnt port) and from there the linux networking stack that performs the rest of the routing.

Its EgressIP for "non ovn managed" interfaces or NICs means ovn isnt configuring the interface and doesnt know about it.

[the remark this answers is not in the document]
Comment (martinkennelly) [reply test/e2e/egressip.go:29]: Does that make sense? @tssurya

[the remark this answers is not in the document]
Comment (martinkennelly) [reply test/e2e/egressip.go:29]: >are we just meaning "the-network-that-is-not-shared-with-br-ex"?

Yeah

File: test/e2e/egressip.go (hunk 2 above)
Comment (martinkennelly) [reply test/e2e/egressip.go:2089]: Done

**Comment by martinkennelly** [comment]
Tested the e2es and they work as long as i disable gateway cleanup (it blindly removes next hops even if it doesnt own them) and also disable pod2pod communication test in the e2es. 
This is being fixed by @tssurya #3825 where we dont need static routes for EIP anymore.

**Comment by martinkennelly** [comment]
These e2es passes against #3825

**Comment by martinkennelly** [comment]
IC mode failed once but then i rebuilt the cluster and it was fine.
I am looping now through rebuilding the cluster and exec the tests and so far after 5 attempts, I havent seen the tests fail. Ill leave it run throughout the day.

**Comment by martinkennelly** [comment]
Should be working now following merging of #3825

**Pull request closed without merging**